uktrade/data-hub-frontend
# Docker compose rework

### Description of change

This reworks docker-compose usage slightly and improves the documentation.
- Locks the redis container to version 3.2 as this is the version we use in production.
- Rewrites mentions to environment variables throughout the README to make things clearer.
- Reworks the docker-compose README section to be a step-by-step guide.
- Advocates mock-sso usage.
- Sets the default environment variables in `sample.env` to point to leeloo/mock-sso docker containers under the same docker-compose project.

This PR should be reviewed in conjunction with the companion leeloo PR: https://github.com/uktrade/data-hub-leeloo/pull/1519

#### How to test

- Please run through the instructions to bring up a docker-compose project locally, including the optional section for setting up leeloo under docker: https://github.com/uktrade/data-hub-leeloo/pull/1519
- Also run through the instructions for setting up mock-sso under docker - this is all in the README which points to the mock-sso README (in this PR: https://github.com/uktrade/mock-sso/pull/2).
- At the end of this, you should have local copies of leeloo, frontend and mock-sso all working together in the same docker-compose project. You should be able to navigate the frontend, be successfully authenticated and see expected data.
